app/assets/javascripts/oxalis/model/sagas/save_saga.js

```diff
@@ -175,6 +175,20 @@ export function compactUpdateActions(updateActionsBatches: Array<Array<UpdateAct
   });
 
 
+  // This part of the code detects deleted trees.
+  // Instead of sending deleteNode/deleteEdge update actions for all nodes of a deleted tree,
+  // just one deleteTree update action is sufficient for the server to delete the tree.
+  // As the deleteTree update action is already part of the update actions if a tree is deleted,
+  // all corresponding deleteNode/deleteEdge update actions can simply be removed.
+  result = result.map((batch) => {
```

Comment:
I agree having result as const would be better and I'll change the code accordingly :)

I'll extract the functions into separate named methods as part of issue #1900, as we start to send grouped update actions then. There won't be any flattening there at the end of `compactUpdateActions` which makes this a little easier.